pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
```

Comment:
I'm not sure that the "always base64" crumminess is worse than the "sometimes base64" crumminess :)

Anyway, I'd like to get #4653 in before any significant retool, please.  I just want it off the balance sheet, and it fixes many of the concerns here.